lib/decimal.dart

```diff
@@ -49,6 +51,18 @@ class Decimal implements Comparable<Decimal> {
     }
   }
 
+  /// Parses [source] as a decimal literal using the provided number formatter and returns its value as [Decimal].
+  static Decimal parseWith(NumberFormat format, String source) => DartDecimalNumberParser(format, source).value!;
```

Comment:
The force cast here fails for invalid values (NaN for example) which in turn also fails to be caught in `tryParse`.

Reply:
I think it is expected for `parseWith` to throw on invalid values, however `tryParseWith` should not, and return `null` when parsing fails.

Reply:
> This is probably because ! is not throwing a FormatException, so the catch clause should widen the exception type it is looking for.

Indeed, but it might be better to always throw a FormatException

Reply:
I agree. Throwing FormatException on NaN will be consistent with `BigInt.parse`. I will change this behaviour. Thanks for your feedback.

Reply:
Fixed in 3.0.1

Reply:
Awesome, thanks!